### PR TITLE
Configure Default Logging for StowRS Tool

### PR DESCRIPTION
**Pull request by dzelemba**
Before this, no log messages were printed.

### CODE REVIEW

File: dcm4che-tool/dcm4che-tool-stowrs/src/main/java/org/dcm4che3/tool/stowrs/StowRS.java [hunk 1]
```diff
@@ -411,6 +413,7 @@ private static void stow(StowRS instance, Attributes metadata, List<String> file
             LOG.info("STOW successful!");
         } catch (Exception e) {
             LOG.error("Exception : " + e.getMessage());
```
Comment (hczedik) [inline dcm4che-tool/dcm4che-tool-stowrs/src/main/java/org/dcm4che3/tool/stowrs/StowRS.java]: Use
`LOG.error("Exception during STOW", e);`
That will automatically print the stack trace and you don't have to do it manually.

**Comment by dzelemba** [comment]
Thanks, fixed. Please take another look.

File: dcm4che-tool/dcm4che-tool-stowrs/pom.xml [hunk 2]
```diff
@@ -50,7 +50,12 @@
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-log4j12</artifactId>
     </dependency>
-        <dependency>
+    <dependency>
+      <groupId>log4j</groupId>
```
Comment (hczedik) [inline dcm4che-tool/dcm4che-tool-stowrs/pom.xml]: No need to pull in log4j explicitly. The dependency above to slf4j-log4j12 should already do it.

File: dcm4che-tool/dcm4che-tool-stowrs/src/main/java/org/dcm4che3/tool/stowrs/StowRS.java [hunk 3]
```diff
@@ -163,6 +164,7 @@ private static Attributes configureKeys(CommandLine cl) {
 
     @SuppressWarnings("unchecked")
     public static void main(String[] args) {
+        BasicConfigurator.configure();
```
Comment (hczedik) [inline dcm4che-tool/dcm4che-tool-stowrs/src/main/java/org/dcm4che3/tool/stowrs/StowRS.java]: Please do not call BasicConfigurator.configure() directly. Instead the configuration should be done using the `log4j.properties` file located in `dcm4che-assembly/src/etc/stowrs/log4j.properties`.
I'm not sure why that doesn't work already correctly. 
@vrindanayak Could you have a look why the stowrs tool behaves differently than the other tools?

**Comment by vrindanayak** [comment]
The tool does log, cannot reproduce the issue. Added the log snippet
````
stowrs --url  http://localhost:8080/dcm4chee-arc/aets/DCM4CHEE/rs/studies   /home/vrinda/Downloads/testdata_2/Äneas^Rüdiger/1.3.6.1.4.1.5962.1.1.0.1.1.1175775772.5723.0 
15:55:32,719 INFO  - added keys for coercion: 

15:55:32,719 INFO  - Check extension of first file only to determine whether STOW is for dicom or non dicom type of objects.
15:55:32,720 INFO  - Storing DICOM objects
15:55:32,768 INFO  - > POST http://localhost:8080/dcm4chee-arc/aets/DCM4CHEE/rs/studies
15:55:32,768 INFO  - > Content-Type: multipart/related; type=application/dicom; boundary=myboundary
15:55:32,768 INFO  - > Accept: application/dicom+xml
15:55:33,174 INFO  - < Content-Length: -1
15:55:33,174 INFO  - < HTTP/1.1 Response: 200 OK
15:55:33,174 INFO  - < Transfer-Encoding: null
15:55:33,174 INFO  - < Content-Type: application/dicom+xml
15:55:33,174 INFO  - < Date: 0
15:55:33,174 INFO  - < Response Content: 
15:55:33,181 DEBUG - <?xml version="1.0" encoding="UTF-8"?><NativeDicomModel xml:space="preserve"><DicomAttribute keyword="RetrieveURL" tag="00081190" vr="UR"><Value number="1">http://localhost:8080/dcm4chee-arc/aets/DCM4CHEE/rs/studies/1.3.6.1.4.1.5962.1.2.0.1175775772.5723.0</Value></DicomAttribute><DicomAttribute keyword="ReferencedSOPSequence" tag="00081199" vr="SQ"><Item number="1"><DicomAttribute keyword="ReferencedSOPClassUID" tag="00081150" vr="UI"><Value number="1">1.2.840.10008.5.1.4.1.1.7</Value></DicomAttribute><DicomAttribute keyword="ReferencedSOPInstanceUID" tag="00081155" vr="UI"><Value number="1">1.3.6.1.4.1.5962.1.1.0.1.1.1175775772.5723.0</Value></DicomAttribute><DicomAttribute keyword="RetrieveURL" tag="00081190" vr="UR"><Value number="1">http://localhost:8080/dcm4chee-arc/aets/DCM4CHEE/rs/studies/1.3.6.1.4.1.5962.1.2.0.1175775772.5723.0/series/1.3.6.1.4.1.5962.1.3.0.1.1175775772.5723.0/instances/1.3.6.1.4.1.5962.1.1.0.1.1.1175775772.5723.0</Value></DicomAttribute></Item></DicomAttribute></NativeDicomModel>
15:55:33,181 INFO  - STOW successful!
````

**Comment by dzelemba** [comment]
Hmm not sure what I'm doing wrong here then, here's my output without the changes:

mvn exec:java -Dexec.mainClass=org.dcm4che3.tool.stowrs.StowRS -Dexec.args="-url <URL> <file>  
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building dcm4che-tool-stowrs 5.13.0
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:1.6.0:java (default-cli) @ dcm4che-tool-stowrs ---
log4j:WARN No appenders could be found for logger (org.dcm4che3.tool.stowrs.StowRS).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.778 s
[INFO] Finished at: 2018-02-21T09:50:27-05:00
[INFO] Final Memory: 15M/487M

**Comment by dzelemba** [comment]
Ok I think I've figured out how to properly invoke the tool now (mvn install & look at the generated dcm4che-assembly outputs). I feel like this should really be documented somewhere, would you be OK if I switched this pull request to add some documentation to the Tools section the top level README that describes how to run the tools?

**Comment by hczedik** [comment]
But if you don't build the tools from source, it is quite easy to run them. Isn't it? Just download the binaries and run one of the tools with the supplied shell script (linux) or bat file (windows).

**Comment by dzelemba** [comment]
Yes if you just download the binaries it's quite easy to run... but if you're making any changes and want to test the tool it took me a bit to figure out how to run it properly (and my first mvn exec solution didn't output any log messages). Maybe a README file in just the dcm4che-tool directory if you're not comfortable with a change to the top-level README?

**Comment by hczedik** [comment]
Okay, I see. I think adding some information to the top-level README on how to run the tools after manual build is a good idea. Maybe in the "Build" section.

**Comment by dzelemba** [comment]
Ok I'll close this pull request then and make the changes in another branch.

**Pull request closed without merging**